### PR TITLE
Query resources by category

### PR DESCRIPTION
**Pull request by styliii**
1. New route: `/v1/workflow/:id/resources`
2. Adds resource serializer, which includes the category from the process

json response looks like this:
```
{
  "data"=>[
    {"id"=>"c2d7-2f8c", "type"=>"resource", "attributes"=>{"title"=>"Such, Such Were the Joys", "link"=>"http://ernser.biz/latina", "updatedAt"=>"2022-11-30T16:00:25.745Z", "categories"=>["Album Advice & Affiliation"]}}, 
    {"id"=>"f5d5-99f3", "type"=>"resource", "attributes"=>{"title"=>"Where Angels Fear to Tread", "link"=>"http://mcclure.info/julian", "updatedAt"=>"2022-11-30T16:00:25.764Z", "categories"=>["Album Advice & Affiliation"]}}, 
    {"id"=>"0236-cd44", "type"=>"resource", "attributes"=>{"title"=>"What's Become of Waring", "link"=>"http://wiza.net/sherise", "updatedAt"=>"2022-11-30T16:00:25.768Z", "categories"=>["Album Advice & Affiliation"]}}, 
    {"id"=>"3ac8-9058", "type"=>"resource", "attributes"=>{"title"=>"Mr Standfast", "link"=>"http://fahey.biz/christian_bradtke", "updatedAt"=>"2022-11-30T16:00:25.784Z", "categories"=>["Finance"]}}, 
    {"id"=>"0275-560b", "type"=>"resource", "attributes"=>{"title"=>"The Far-Distant Oxus", "link"=>"http://schultz.name/jerrold_ziemann", "updatedAt"=>"2022-11-30T16:00:25.789Z", "categories"=>["Finance"]}}, 
    {"id"=>"62ae-1e3c", "type"=>"resource", "attributes"=>{"title"=>"Things Fall Apart", "link"=>"http://sauer.co/homer.green", "updatedAt"=>"2022-11-30T16:00:25.793Z", "categories"=>["Finance"]}}]
}
```

### CODE REVIEW

File: app/controllers/v1/workflow/workflows_controller.rb [hunk 1]
```diff
@@ -6,4 +6,17 @@ def show
...
     render json: V1::Workflow::WorkflowSerializer.new(@workflow, include: ['processes'])
   end
+
+  def resources
+    @workflow = Workflow::Instance::Workflow.find_by!(external_identifier: params[:workflow_id])
+    process_ids = @workflow.processes.pluck(:id)
+    instance_step_ids = Workflow::Instance::Step.where(process_id: process_ids).pluck(:id)
+    definition_step_ids = Workflow::Instance::Step.where(process_id: process_ids).pluck(:definition_id)
+
+    ## document could either be from instance or definition
```
Comment (keithtom) [inline app/controllers/v1/workflow/workflows_controller.rb:16]: Good catch, didn't realize this.

File: spec/models/workflow/instance/process_spec.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  let(:process) { create(:workflow_instance_process) }
+  describe "when the instance does not have its own categories" do
+    it "fetches categories from the definition" do
+      expect(process.category_list).to be_empty
+      process.definition.category_list.add("Finance")
+      process.definition.save!
+      expect(process.reload.category_list).to_not be_empty
+    end
+  end
+end
```
Comment (keithtom) [inline spec/models/workflow/instance/process_spec.rb:13]: Do you mind testing the positive case too?  When it does have its own categories, it uses that over definition?

**Review by keithtom — APPROVED** [review]
LGTM!

File: spec/models/workflow/instance/process_spec.rb (hunk 2 above)
Comment (styliii) [reply spec/models/workflow/instance/process_spec.rb:5]: Good call. When i tried to test the positive case, i realized my "clever" method was providing a bit too much magic. There was no else statement, and more dangerously, you'd always be adding categories to the definition.  `#category_list.add` would return and add new categories to the definition's category list. 

I created a helper function, `get_categories` and put it into the concern.

**Review by keithtom — APPROVED** [review]
PR seems fine to me.  I'm not 100% sure on the concern, doesn't seem that much more readable than putting it directly in the serializer unless you know it'll be used elsewhere.  But pretty minor issue for now.

**Pull request merged**